code/go/0chain.net/smartcontract/zcnsc/models.go

```diff
@@ -0,0 +1,497 @@
+package zcnsc
+
+import (
+	"0chain.net/chaincore/chain"
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/config"
+	"0chain.net/chaincore/state"
+	"0chain.net/chaincore/tokenpool"
+	"0chain.net/chaincore/transaction"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	// "0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+	"0chain.net/core/datastore"
+	"0chain.net/core/encryption"
+	"0chain.net/core/util"
+)
+
+var (
+	allAuthorizerKey = ADDRESS + encryption.Hash("all_authorizers")
+)
+
+type globalNode struct {
+	ID                 string        `json:"id"`
+	MinMintAmount      state.Balance `json:"min_mint_amount"`
+	PercentAuthorizers float64       `json:"percent_authorizers"`
+	MinBurnAmount      int64         `json:"min_burn_amount"`
+	MinStakeAmount     int64         `json:"min_stake_amount"`
+	BurnAddress        string        `json:"burn_address"`
+	MinAuthorizers     int64         `json:"min_authorizers"`
+}
+
+func (gn *globalNode) GetKey() datastore.Key {
+	return ADDRESS + gn.ID
+}
+
+func (gn *globalNode) GetHash() string {
+	return util.ToHex(gn.GetHashBytes())
+}
+
+func (gn *globalNode) GetHashBytes() []byte {
+	return encryption.RawHash(gn.Encode())
+}
+
+func (gn *globalNode) Encode() []byte {
+	buff, _ := json.Marshal(gn)
+	return buff
+}
+
+func (gn *globalNode) Decode(input []byte) error {
+	err := json.Unmarshal(input, gn)
+	return err
+}
+
+func (gn *globalNode) save(balances cstate.StateContextI) (err error) {
+	_, err = balances.InsertTrieNode(gn.GetKey(), gn)
+	return
+}
+
+func getGlobalSavedNode(balances cstate.StateContextI) (*globalNode, error) {
+	gn := &globalNode{ID: ADDRESS}
+	gv, err := balances.GetTrieNode(gn.GetKey())
+	if err != nil {
+		return gn, err
+	}
+	_ = gn.Decode(gv.Encode())
+	return gn, err
+}
+
+// getGlobalNode - returns global node
+func getGlobalNode(balances cstate.StateContextI) *globalNode {
+	gn, err := getGlobalSavedNode(balances)
+	if err == nil {
+		return gn
+	}
+
+	gn.MinMintAmount = state.Balance(config.SmartContractConfig.GetInt("smart_contracts.zcn.min_mint_amount"))
+	gn.PercentAuthorizers = config.SmartContractConfig.GetFloat64("smart_contracts.zcn.percent_authorizers")
+	gn.MinAuthorizers = config.SmartContractConfig.GetInt64("smart_contracts.zcn.min_authorizers")
+	gn.MinBurnAmount = config.SmartContractConfig.GetInt64("smart_contracts.zcn.min_burn_amount")
+	gn.MinStakeAmount = config.SmartContractConfig.GetInt64("smart_contracts.zcn.min_stake_amount")
+	gn.BurnAddress = config.SmartContractConfig.GetString("smart_contracts.zcn.burn_address")
```

Comment:
I am confused. The global node is retrieved from the blockchain, then all the fields are overwritten. Why the blockchain access?

Reply:
@Sriep This is repeated after many other places, for example, in miners, the globalNode after fetching is being updated according to configuration

Reply:
Sorry, I think its ok, leave as is.